docs/tutorial.md

```diff
@@ -18,6 +18,86 @@ The wizard will also ask if you'd like it setup any additional metadata files su
 > Ref: See [`zigmod init`](./commands/init.md) for more info.
 
 ---
+## Understanding `zig.mod`
+
+Running `zigmod init` will ask if You want to create an application or a
+library. Based on You answer the resulting `zig.mod` file have two types.
+
+### `zig.mod` for an application
+
+```yaml
+id: ucst208pgxu7c1slqop2ng9llcjy4369zmupp21wxjvv0nib
+name: my-package
+license: None
+description: None
+root_dependencies:
+  - src: git https://github.com/jecolon/ziglyph
+```
+
+In this case You can use `@import("ziglyph")` to import the package and use it.
+
+### `zig.mod` for a library
+
+```yaml
+id: ucst208pgxu7c1slqop2ng9llcjy4369zmupp21wxjvv0nib
+name: my-package
+main: src/lib.zig
+license: None
+description: None
+dependencies:
+  - src: git https://github.com/jecolon/ziglyph
+```
+
+Here I assume that You initialized Your project with `zig init-exe` or
+`zig init-lib`, so Your `build.zig` the `addExecutable()` or
+`addStaticLibrary()` call points to `src/main.zig`.
+
+In this case You can `@import("ziglyph")` in `src/lib.zig`, but not in
+`src/main.zig`. However You can `@import("my-package")` in `src/main.zig`.
+
+Please note, that using `zig init-exe` or `zig init-lib` for creating a project
+does not mean that You have to use `zigmod init` in application or library mode.
+Creating a `zig.mod` file with proper `name` and `main` only makes Your package
+importable by other projects. This is usually not needed with applications, but
+quite useful in case of libraries; hence the name.
+
+### Zig and Zigmod package handling explained
+
+Zig packages can only `@import` **direct** children. The include tree is
+something like this:
+
+```
+root
+  A
+    C
+  B
+  D
+    E
+```
+
+`root` can `@import` `A`, `B` and `D`, but `E` can be imported only from `D`.
+`B` can not import anything.
+(Everybody can import relative files, this limitation is only for packages.)
+
+In library `zig.mod` example the tree looks like this:
+
+```
+root (src/main.zig)
+  my-package (src/lib.zig)
+    ziglyph (separate package)
+```
+
+`root` is always set in `build.zig` using `addExecutable()` or
+`addStaticLibrary()` calls.
+
+It is very important to understand, that `my-package` is _below_ `root`, as
+`name/main` in `zig.mod` will create a separate package (it will not be chained to
+`root`).
+
+If You want to use the dependencies from `root`, then make sure that Your
+`zig.mod` does _not_ contain a `main` line **and** use `root_dependencies`
+instead of `dependencies`. This is what `zigmod init` does in application mode.
```

Comment:
the use of `root_dependencies:` and `dependencies:` is not mutually exclusive, however you correctly point out that `dependencies:` does require the use of `main:` (this simultaneous app/package setup can be handy and taken advantage of by Zigmod itself)